ql/cashflows/couponpricer.hpp

```diff
@@ -80,12 +84,31 @@ namespace QuantLib {
         Handle<OptionletVolatilityStructure> capletVol_;
     };
 
-    //! Black-formula pricer for capped/floored Ibor coupons
+    /*! Black-formula pricer for capped/floored Ibor coupons
+        References for timing adjustments
+        Black76             Hull, Options, Futures and other
+                            derivatives, 4th ed., page 550
+        BivariateLognormal  http://ssrn.com/abstract=2170721
+        The bivariate lognormal adjustment implementation is
+        still considered experimental */
     class BlackIborCouponPricer : public IborCouponPricer {
       public:
-        BlackIborCouponPricer(const Handle<OptionletVolatilityStructure>& v =
-                                        Handle<OptionletVolatilityStructure>())
-        : IborCouponPricer(v) {};
+        enum TimingAdjustment { Black76, BivariateLognormal };
+        BlackIborCouponPricer(
+            const Handle< OptionletVolatilityStructure > &v =
+                Handle< OptionletVolatilityStructure >(),
+            const TimingAdjustment timingAdjustment = Black76,
+            const Handle< Quote > correlation =
+                Handle< Quote >(boost::make_shared< SimpleQuote >(1.0)))
```

Comment:
Yes, I think it’s sensible, because the rates in question are usually close to each other on the curve (except for large payment delays), so rather highly correlated; there is also a consistency requirement that rho goes to 1.0 if the payment time approaches the natural payment time. So I think it’s a good default value, if the user can not provide one.

> Am 24.02.2016 um 14:55 schrieb Luigi Ballabio <notifications@github.com>:
> 
> In ql/cashflows/couponpricer.hpp <https://github.com/lballabio/QuantLib/pull/56#discussion_r53940671>:
> 
> >      class BlackIborCouponPricer : public IborCouponPricer {
> >        public:
> > -        BlackIborCouponPricer(const Handle<OptionletVolatilityStructure>& v =
> > -                                        Handle<OptionletVolatilityStructure>())
> > -        : IborCouponPricer(v) {};
> > +        enum TimingAdjustment { Black76, BivariateLognormal };
> > +        BlackIborCouponPricer(
> > +            const Handle< OptionletVolatilityStructure > &v =
> > +                Handle< OptionletVolatilityStructure >(),
> > +            const TimingAdjustment timingAdjustment = Black76,
> > +            const Handle< Quote > correlation =
> > +                Handle< Quote >(boost::make_shared< SimpleQuote >(1.0)))
> Do you think 1 is a sensible default value for correlation when not given explicitly, or was it just to avoid an empty handle?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/lballabio/QuantLib/pull/56/files#r53940671>.
>